src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -23,10 +24,10 @@ namespace System.ComponentModel
     /// </summary>
     internal sealed partial class ReflectTypeDescriptionProvider : TypeDescriptionProvider
     {
-        // Hastable of Type -> ReflectedTypeData. ReflectedTypeData contains all
+        // ReflectedTypeData contains all
         // of the type information we have gathered for a given type.
         //
-        private Hashtable? _typeData;
+        private ConcurrentDictionary<Type, ReflectedTypeData>? _typeData;
```

Comment:
I'm not sure I can trivially prove the changes in this file are correct. Can we revert this file's changes?